### PR TITLE
Adding types to Webhooks

### PR DESCRIPTION
**Pull request by dliu-workos**

### CODE REVIEW

File: src/main/kotlin/com/workos/webhooks/models/WebhookJsonDeserializer.kt [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import com.workos.sso.models.Connection
+
+/**
+ * Custom JSON deserializer for webhook events.
+ */
+class WebhookJsonDeserializer : JsonDeserializer<Webhook>() {
+  private val mapper = ObjectMapper().registerModule(KotlinModule.Builder().build())
+  override fun deserialize(jp: JsonParser?, ctxt: DeserializationContext?): Webhook {
+    val rootNode = jp?.codec?.readTree<TreeNode>(jp)
+    val id = mapper.readValue(rootNode?.get("id")?.traverse(), String::class.java)
```
Comment (dliu-workos) [inline src/main/kotlin/com/workos/webhooks/models/WebhookJsonDeserializer.kt:21]: There's gotta be a better way to get the string value for this field...

File: src/main/kotlin/com/workos/directorysync/models/PreviousAttributes.kt [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import com.fasterxml.jackson.annotation.JsonCreator
+import com.fasterxml.jackson.annotation.JsonProperty
+
+/**
+ * An object representing the previous attributes of names and values
+ * that have changed and their previous values.
+ *
+ * @param previousAttributes The object containing the previous keys and values for changes.
+ */
+data class PreviousAttributes @JsonCreator constructor(
```
Comment (dliu-workos) [inline src/main/kotlin/com/workos/directorysync/models/PreviousAttributes.kt:12]: unused-- trying to see how we can compose `User` and `UserWithPreviousAttributes`

Comment (dliu-workos) [reply src/main/kotlin/com/workos/directorysync/models/PreviousAttributes.kt:12]: This is not as straightforward as I thought. Kotlin doesn't allow you to inherit constructors. Removing for now and will figure it out on the next iteration.

Comment (dliu-workos) [reply src/main/kotlin/com/workos/directorysync/models/PreviousAttributes.kt:12]: will work on this in the next iteration

File: src/main/kotlin/com/workos/directorysync/models/DirectoryState.kt [hunk 3]
```diff
@@ -8,6 +8,18 @@ import com.fasterxml.jackson.annotation.JsonValue
  * @param state The Directory State string value.
  */
 enum class DirectoryState(@JsonValue val state: String) {
+  /**
+   * The directory is connected to an external provider
+   */
+  Active("active"),
+  /**
+   * The
```
Comment (dliu-workos) [inline src/main/kotlin/com/workos/directorysync/models/DirectoryState.kt:16]: oops

File: src/main/kotlin/com/workos/webhooks/models/WebhookJsonDeserializer.kt (hunk 1 above)
Comment (dliu-workos) [reply src/main/kotlin/com/workos/webhooks/models/WebhookJsonDeserializer.kt:25]: Will leave it as is for now.

**Review by robframpton — APPROVED** [review]

**Comment by robframpton** [comment]
Have we tested this in the Java example app yet?

**Comment by dliu-workos** [comment]
> Have we tested this in the Java example app yet?

Yep! I'm doing that now.
https://github.com/workos-inc/java-example-applications/pull/16

**Pull request merged**